src/main/java/org/apache/sling/jcr/repoinit/impl/RepositoryInitializerFactory.java

```diff
@@ -78,14 +79,22 @@ public class RepositoryInitializerFactory implements SlingRepositoryInitializer
 
     private final Logger log = LoggerFactory.getLogger(getClass());
 
+    private static final String METRIC_REPOINIT_FAILED = RepositoryInitializerFactory.class.getName() + ".failed";
 
     @Reference
     private RepoInitParser parser;
 
     @Reference
     private JcrRepoInitOpsProcessor processor;
+    
+    @Reference
+    MetricsService metrics;
 
     private RepositoryInitializerFactory.Config config;
+    
+    // assume that repoinit succeeds ... and just this to true if it fails
+    private boolean aRepoInitStatementFailed = false;
```

Comment:
I think this field should be `volatile` (or another mechanism to ensure visibility should be used). IMHO there is a high probability that the metric is written and read from different threads.